src/browser.ts

```diff
@@ -88,31 +93,26 @@ function hasLegacyDestinations(settings: LegacySettings): boolean {
   )
 }
 
-async function flushBuffered(analytics: Analytics): Promise<void> {
-  const wa = window.analytics
-  const buffered =
-    // @ts-expect-error
-    wa && wa[0] ? [...wa] : []
-
-  for (const [operation, ...args] of buffered) {
-    if (
-      // @ts-expect-error
-      analytics[operation] &&
-      // @ts-expect-error
-      typeof analytics[operation] === 'function'
-    ) {
-      if (operation === 'addSourceMiddleware') {
-        // @ts-expect-error
-        await analytics[operation].call(analytics, ...args)
-      } else {
-        // flush each individual event as its own task, so not to block initial page loads
-        setTimeout(() => {
-          // @ts-expect-error
-          analytics[operation].call(analytics, ...args)
-        }, 0)
-      }
+function flushBuffered(analytics: Analytics, buffer: PreInitMethodCallBuffer) {
+  const callBufferedAnalyticsMethod = (methodCall: PreInitMethodCall) => {
+    const { method } = methodCall
+    // this guard is probably not needed.
+    if (typeof analytics[method] !== 'function') {
+      return console.warn(
+        `invariant error: method call "${method}" does not exist on analytics instance: ${analytics}`
+      )
+    }
+    if (method === 'addSourceMiddleware') {
+      void buffer.callMethod(analytics, methodCall)
```

Comment:
"void" is something that eslint wants for expressions that return a promise that we will explicitly _not be handling_. Otherwise, you get `no-floating-promises`.

Reply:
Hmm, that's interesting, but it looks like it also prevents us from handling any errors that might be thrown by the promise.

I just did a quick test [here](https://jsfiddle.net/okm2fh5y/2/) and when I look at the dev tools console after running I see the uncaught error (in promise) thrown. In node.js this would cause the process to exit. If we truly want to not worry about errors from some promises we may want to explicitly ignore them or log them.

Reply:
> Hmm, that's interesting, but it looks like it also prevents us from handling any errors that might be thrown by the promise.

Oops, good callout. I've tried to preserve the existing logic and I made a mistake looking at the old code; this should be awaited!

On that note, the existing logic allows for unhandled promise rejections:

https://github.com/segmentio/analytics-next/blob/aa423cc828fc9da0a4df6603dbda9ab3978db698/src/browser.ts#L109-L112

^ should we handle these?

> I just did a quick test [here](https://jsfiddle.net/okm2fh5y/2/) and when I look at the dev tools console after running I see the uncaught error (in promise) thrown. In node.js this would cause the process to exit.

Yep, 'void' is like a comment!
```ts
void Promise.reject(123) // VM364:1 Uncaught (in promise) 123 
```
That's a good point about nodejs --  I guess the fact that this is browser code means that we haven't had any complaints.

Reply:
> ^ should we handle these?

Oh nice catch! I think so yes. For what it's worth, I think we do make it pretty hard for most of our APIs to actually throw an error to the end-user, though I wouldn't bet my node process on that!

Reply:
I added error handling to the `callMethod` function, and have some tests around it.